### PR TITLE
[Backport 2.23.x] [GEOS-10949] Control remote resources accessed by GeoServer

### PR DESCRIPTION
**Pull request by canmehteroglu**
[GEOS-10949](https://osgeo-org.atlassian.net/browse/GEOS-10949)

Backport of url checks but disabled by default.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by sikeoka** [comment]
@aaime, @jodygarnett: I missed this on the main PR but WPS requests can actually make HTTP requests to two different URLs.  The check was added to the first URL but is missing on the second URL :
https://github.com/geoserver/geoserver/blob/main/src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/RemoteRequestInputProvider.java#L174

**Comment by sikeoka** [comment]
Do you have any thoughts on my two comments?

**Comment by aaime** [comment]
@sikeoka the original PR had a GSIP, it has been discussed, implemented and merged over the course of the last two months, the PR has been open for two weeks, and has landed on main over a month ago. 
You did not say anything for all that time, then pop up and then pretend to hear back within the day....

That said, yes, we are looking into it and will make a PR to strengthen the regex. If you have a quick suggestion on how to implement it, we're all ears (or if you'd like to make such PR yourself, that would be even better).

**Comment by sikeoka** [comment]
I wasn't sure if backport PRs are an appropriate place for discussions or if you would have preferred a mailing list.  I try to review any hz-cluster, jdbcconfig and ysld PRs (I'm pretty sure I'm a maintainer for at least the first two) but, other than that, I'm usually not reviewing every PR unless I am specifically pinged on the PR.  Every 3-6 months or so, I will sync local copies of the repos and review every single commit for potential security issues and this one happened to have an open PR.

**Comment by aaime** [comment]
We gladly accept all the help we can get. You checking every commit once every 3-6 months is just fine. Assuming we'll reply back to your comments within the day, is not.

File: doc/en/user/source/security/images/regex.png
[the remark this answers is not in the document]
Comment (jodygarnett) [reply doc/en/user/source/security/images/regex.png:1]: Can you take this up as a new issue, addressing your concern will have to be done on main first as this is only a backport PR.

**Comment by jodygarnett** [comment]
Resolving conflict to see if this can be applied ahead of next release

**Comment by jodygarnett** [comment]
> @aaime, @jodygarnett: I missed this on the main PR but WPS requests can actually make HTTP requests to two different URLs. The check was added to the first URL but is missing on the second URL :
> https://github.com/geoserver/geoserver/blob/main/src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/RemoteRequestInputProvider.java#L174

I have added this check to https://github.com/geoserver/geoserver/pull/6916 - please review

**Comment by jodygarnett** [comment]
Commit 9b21a8d887c969d21f0aac4b522e5fc23cfb9989 is ready to be included in this PR

**Pull request merged**